### ISSUE TITLE
Total params is twice as large

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#26]
When loading a trained model that has been wrapped using `GAModelWrapper`, we get:
```
Model: "ga_model_wrapper"
_______________________________________________________________________________
[... Layer (type) info stuff ...]
================================================
[...]
================================================
Total params: 171,700,165
Trainable params: 85,850,082
Non-trainable params: 85,850,083
_______________________________________________________________________________
```

The problem seems to be that the model contains double the number of params, which likely comes from that all layers have been renamed, such that two copies of the same model exist in the graph.

This might also impact GPU memory allocation.

The expected behaviour is for the layer/weight names to stay the same.

**Comment by andreped** [comment]
This is likely related to https://github.com/andreped/GradientAccumulator/issues/25

**Comment by andreped** [comment]
We may have to do something like [this](https://github.com/keras-team/keras/blob/v2.12.0/keras/mixed_precision/loss_scale_optimizer.py#L873).

**Comment by tno123** [comment]
Unfortunately, double parameters during gradient accumulation is expected behavior due to the accumulation of gradients for each parameter separately using the gradient_accumulation variables.
